codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -56,6 +57,29 @@
 
 abstract class AbstractHttp2StreamChannel extends DefaultAttributeMap implements Http2StreamChannel {
 
+    static final ChannelHandler NOOP_UPGRADE_HANDLER = new ChannelInboundHandlerAdapter() {
+
+        @Override
+        public boolean isSharable() {
+            return true;
+        }
+
+        @Override
+        public void channelRead(ChannelHandlerContext ctx, Object msg) {
+            ReferenceCountUtil.release(msg);
```

Comment:
Why does this ignore/throw away the message, event, and exceptions instead of letting them go to the end of the pipeline?

Reply:
because the user has no way to access the Channel and so the pipeline to add handlers by himself / herself ?